src/libs/actions/TaxRate.ts

```diff
@@ -486,6 +487,33 @@ function renamePolicyTax(policyID: string, taxID: string, newName: string) {
 function setPolicyTaxCode(policyID: string, oldTaxCode: string, newTaxCode: string) {
     const policy = allPolicies?.[`${ONYXKEYS.COLLECTION.POLICY}${policyID}`];
     const originalTaxRate = {...policy?.taxRates?.taxes[oldTaxCode]};
+    const customUnits = Object.values(policy?.customUnits ?? {});
+    const optimisticCustomUnit = {
+        customUnits: {
+            ...customUnits.reduce((units, customUnit) => {
```

Comment:
Maybe I'm misunderstanding but wouldn't this be better as a map instead of a reduce? Same with the rates.reduce below

Reply:
customUnits is an array though right? Could do something here like
```js
customUnits: customUnits.map((customUnit) => {
	let newUnit = customUnit;
	// modify rates
	return newUnit
})
```
Seems a little cleaner? And could use a forEach for the rates object?

what do you think @rushatgabhane

Reply:
oh yeah, fair. Okay then I think this works for me! Thank you for the explanation, that was helpful!